Kernel/Storage/IntelAHCIPort.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2021, Alexander Richards <electrodeyt@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/CommandLine.h>
+#include <Kernel/Storage/ATA.h>
+#include <Kernel/Storage/IntelAHCIPort.h>
+#include <Kernel/Storage/SATADiskDevice.h>
+#include <Kernel/VM/TypedMapping.h>
+
+namespace Kernel {
+
+NonnullRefPtr<AHCIPort> IntelAHCIPort::create(const AHCIPortHandler& handler, volatile AHCI::PortRegisters& registers, u32 port_index)
+{
+    return adopt_ref(*new IntelAHCIPort(handler, registers, port_index));
+}
+
+IntelAHCIPort::IntelAHCIPort(const AHCIPortHandler& handler, volatile AHCI::PortRegisters& regs, u32 port_index)
+    : AHCIPort(handler, regs, port_index)
+{
+    PCI::Address controller_address = handler.hba_controller()->pci_address();
+    if(PCI::get_id(controller_address).device_id == 0x3b22) {
+        m_has_reset_quirk = true;
+    }
+}
+
+bool IntelAHCIPort::initialize(ScopedSpinLock<SpinLock<u8>>& main_lock)
```

Comment:
This feels like a duplication of `AHCIPort::initialize` method. Maybe we can do it in a better way, like do specific stuff for the Intel AHCI controller ports and then call the more generic `AHCIPort::initialize` method?